proposals/0341-v0-account-compression.md

```diff
@@ -0,0 +1,275 @@
+but the complexity isn't currently necessary.
+
+### reduce and fix hot-set size with a chili peppers-like approach
+
+**Pros**: keeps all data on-chain, no need for users to manually recover old
+accounts
+**Cons**: doesn't reduce total state size so snapshots remain large and
+rent remains high
+
+Chili peppers has other applications but may not be necessary if account
+compression can reduce the global state size sufficiently.
+
+### Conclusion
+
+The proposed hash-based approach provides the optimal balance of storage
+savings, performance predictability, and implementation complexity.
+
+## Impact
+
+### DApp and Wallet Developers
+
+- include checks for account compression status in program interaction
+  workflow
+- add instructions to transactions for account recovery when appropriate
+- collisions on the `pubkey_hash` are extremely unlikely but not impossible
+  due to using 10-bytes. In this case, the user or developer will need to use
+  a different address.
+
+### Validators
+
+- **Memory/Storage savings**: If enough accounts are compressed, a
+  significant reduction in disk/memory usage is expected
+- **Network impact**: Reduced snapshot sizes improve sync times
+
+### Core Contributors
+
+- **Implementation scope**: Changes required in runtime, banking, and
+  snapshot systems
+- **Testing requirements**: Comprehensive testing of compression/recovery cycles
+- **Monitoring needs**: New metrics for compression performance, compressed
+  state size, rate, etc
+
+## Security Considerations
+
+### Data Integrity
+
+- **Hash verification**: All recovery operations verify data integrity via
+  hash comparison
+- **Atomic operations**: Compression/recovery operations are atomic to
+  ensure consistent state across the cluster
+
+### Attack Vectors
+
+- **Hash collision attacks**: collisions on the `data_hash` would allow for
+  introducing arbitrary data into the account state. SHA-256 provides
+  sufficient collision resistance. Collisions on the `pubkey_hash` are not
+  concerning since they're extremely rare and only prevent the new account
+  from being created.
+
+## Backwards Compatibility
+
+This feature introduces breaking changes:
+
+### Bank Hash Changes
+
+- **Impact**: Bank hash calculation includes compressed accounts, affecting
+  consensus
+- **Mitigation**: Feature gate activation ensures all validators adopt
+  simultaneously
+
+### Snapshot Format
+
+- **Impact**: New snapshot format with compressed accounts section
+- **Mitigation**: Version-aware snapshot loading with backward compatibility
+  for old snapshots
+
+### Account Creation Behavior  
+
+- **Impact**: Account creation fails if `SHA_256(pubkey)[0:10]` exists in
+  compressed set
+- **Mitigation**: if the account corresponding to the pubkey was previously
+  compressed it must be recovered rather than recreated. if a collision has
+  occured a different pubkey must be used. RPCs and good errors can provide
```

Comment:
> RPCs and good errors can provide the relevant info

This is incredibly important to get right. It we don't outline exactly all the edgecases and how this works for developers, it'll be a nightmare for anyone to integrate with. We should consider putting more detail either in this SIMD on how these edge cases and errors will be displayed, or have a separate document outlining how we can properly setup the right infrastructure to handle the expected behavior with compressed state.